reana_commons/publisher.py

```diff
@@ -24,54 +24,60 @@
 import json
 import logging
 
-import pika
+from kombu import Connection, Exchange, Queue
 
-from .config import (BROKER_PASS, BROKER_PORT, BROKER_URL, BROKER_USER,
-                     EXCHANGE, ROUTING_KEY, STATUS_QUEUE)
+from .config import (BROKER, BROKER_PASS, BROKER_PORT, BROKER_URL, BROKER_USER,
+                     EXCHANGE, MQ_DEFAULT_EXCHANGE, MQ_DEFAULT_QUEUE,
+                     MQ_DEFAULT_ROUTING_KEY, MQ_DEFAULT_SERIALIZER,
+                     MQ_PRODUCER_MAX_RETRIES, ROUTING_KEY, STATUS_QUEUE)
 
 
-class Publisher:
+class WorkflowStatusPublisher():
     """Progress publisher to MQ."""
 
-    def __init__(self):
+    def __init__(self, connection=None, routing_key=None, exchange=None):
         """Initialise the Publisher class."""
-        self.broker_credentials = pika.PlainCredentials(BROKER_USER,
-                                                        BROKER_PASS)
-        self._params = pika.connection.ConnectionParameters(
-            BROKER_URL, BROKER_PORT, '/', self.broker_credentials)
-        self._conn = None
-        self._channel = None
+        self._routing_key = routing_key or MQ_DEFAULT_ROUTING_KEY
+        self._exchange = Exchange(name=exchange or MQ_DEFAULT_EXCHANGE,
+                                  type='direct')
+        self._queue = Queue(MQ_DEFAULT_QUEUE, durable=False,
+                            exchange=MQ_DEFAULT_EXCHANGE,
+                            routing_key=self._routing_key)
+        self._connection = connection or Connection(BROKER)
+        self.producer = self._producer()
```

Comment:
Maybe the method could be renamed so this line becomes: `self.producer = self.get_producer()`